ui/lib/kubernetes/addon/components/page/credentials.hbs

```diff
@@ -0,0 +1,144 @@
+<PageHeader as |p|>
+  <p.top>
+    <nav class="breadcrumb" aria-label="breadcrumbs">
+      <ul>
+        <li>
+          <span class="sep">/</span>
+          <LinkTo @route="overview">{{@backend}}</LinkTo>
+        </li>
+        <li>
+          <span class="sep">/</span>
+          <LinkTo @route="roles">roles</LinkTo>
+        </li>
+        <li>
+          <span class="sep">/</span>
+          <LinkTo @route="roles.role.details">{{@role.name}}</LinkTo>
+        </li>
+        <li>
+          <span class="sep">/</span>
+          <span>credentials</span>
+        </li>
+      </ul>
+    </nav>
+  </p.top>
+  <p.levelLeft>
+    <h1 class="title is-3 has-bottom-margin-2" data-test-credentials-header>
+      {{if this.credentials "Credentials" "Generate credentials"}}
+    </h1>
+  </p.levelLeft>
+</PageHeader>
+
+{{#if this.credentials}}
+  <div class="box is-sideless is-fullwidth is-marginless has-bottom-padding-l" data-test-credentials-details>
+    <AlertBanner
+      @class="is-marginless"
+      @type="warning"
+      @title="Warning"
+      @message="You won't be able to access these credentials later, so please copy them now."
+    />
+    <InfoTableRow @label="Service account token">
+      <MaskedInput
+        @value={{this.credentials.service_account_token}}
+        @name="Service Account Token"
+        @displayOnly={{true}}
+        @allowCopy={{true}}
+      />
+    </InfoTableRow>
+
+    <InfoTableRow @label="Namespace">
+      <CopyButton
+        class="button is-transparent is-flex-v-centered"
+        @clipboardText={{this.credentials.service_account_namespace}}
+        @buttonType="button"
+        @success={{action (set-flash-message "Namespace copied")}}
+      >
+        <Icon class="has-text-grey-light" @name="clipboard-copy" aria-label="Copy" />
+        <span class="has-font-weight-normal title is-7">{{this.credentials.service_account_namespace}}</span>
+      </CopyButton>
+    </InfoTableRow>
+
+    <InfoTableRow @label="Service account name">
+      <CopyButton
+        class="button is-transparent is-flex-v-centered"
+        @clipboardText={{this.credentials.service_account_name}}
+        @buttonType="button"
+        @success={{action (set-flash-message "Service account name copied")}}
+      >
+        <Icon class="has-text-grey-light" @name="clipboard-copy" aria-label="Copy" />
+        <span class="has-font-weight-normal title is-7">{{this.credentials.service_account_name}}</span>
+      </CopyButton>
+    </InfoTableRow>
```

Comment:
It looks like the `InfoTableRow` component has an `addCopyButton` argument so you could use that and pass in the value to achieve the same thing here I think.

Reply:
That's really useful! Thanks for pointing that out!